src/cluster_linearize.h

```diff
@@ -184,6 +157,34 @@ class DepGraph
         }
     }
 
+    /** Modify this transaction graph, adding multiple parents to a specified child.
+     *
+     * The behavior is identical to repeatedly calling AddDependency for each element of parents,
+     * but is more efficient.
+     *
+     * Complexity: O(N) where N=TxCount().
+     */
+    void AddDependencies(const SetType& parents, ClusterIndex child) noexcept
+    {
```

Comment:
Was reconstructing graphs incorrectly leading to inconsistent sets. Better to fail here early.

```Suggestion
    {
         // We can't add relationships to holes
        Assume(parents.IsSubsetOf(m_used));
```
a check like this for `AddDependency` would also be good:

`Assume(SetType::Singleton(parent).IsSubsetOf(m_used));`